contracts/scripts/foundry/InitializeL2ScrollOwner.s.sol

```diff
@@ -0,0 +1,237 @@
+        bytes4[] memory _selectors;
+
+        // delay 7 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = L2TxFeeVault.updateMinWithdrawAmount.selector;
+        owner.updateAccess(L2_TX_FEE_VAULT_ADDR, _selectors, TIMELOCK_7DAY_DELAY_ROLE, true);
+    }
+
+    function configWhitelist() internal {
+        bytes4[] memory _selectors;
+
+        // delay 1 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = Whitelist.updateWhitelistStatus.selector;
+        owner.updateAccess(L2_WHITELIST_ADDR, _selectors, TIMELOCK_1DAY_DELAY_ROLE, true);
+    }
+
+    function configL2ScrollMessenger() internal {
+        bytes4[] memory _selectors;
+
+        // no delay, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = ScrollMessengerBase.setPause.selector;
+        owner.updateAccess(L2_SCROLL_MESSENGER_PROXY_ADDR, _selectors, SCROLL_MULTISIG_NO_DELAY_ROLE, true);
+    }
+
+    function configL2GatewayRouter() internal {
+        bytes4[] memory _selectors;
+
+        // delay 7 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = L2GatewayRouter.setERC20Gateway.selector;
+        owner.updateAccess(L2_GATEWAY_ROUTER_PROXY_ADDR, _selectors, TIMELOCK_1DAY_DELAY_ROLE, true);
+    }
+
+    function configL2CustomERC20Gateway() internal {
+        bytes4[] memory _selectors;
+
+        // delay 1 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = L2CustomERC20Gateway.updateTokenMapping.selector;
+        owner.updateAccess(L2_CUSTOM_ERC20_GATEWAY_PROXY_ADDR, _selectors, TIMELOCK_1DAY_DELAY_ROLE, true);
+    }
+
+    function configL2ERC721Gateway() internal {
+        bytes4[] memory _selectors;
+
+        // delay 1 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = L2ERC721Gateway.updateTokenMapping.selector;
+        owner.updateAccess(L2_ERC721_GATEWAY_PROXY_ADDR, _selectors, TIMELOCK_1DAY_DELAY_ROLE, true);
+    }
+
+    function configL2ERC1155Gateway() internal {
+        bytes4[] memory _selectors;
+
+        // delay 1 day, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = L2ERC1155Gateway.updateTokenMapping.selector;
+        owner.updateAccess(L2_ERC1155_GATEWAY_PROXY_ADDR, _selectors, TIMELOCK_1DAY_DELAY_ROLE, true);
+    }
+
+    function configETHRateLimiter() internal {
+        bytes4[] memory _selectors;
+
+        // no delay, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = ETHRateLimiter.updateTotalLimit.selector;
+        owner.updateAccess(L2_ETH_RATE_LIMITER_ADDR, _selectors, SCROLL_MULTISIG_NO_DELAY_ROLE, true);
+    }
+
+    function configTokenRateLimiter() internal {
+        bytes4[] memory _selectors;
+
+        // no delay, scroll multisig
+        _selectors = new bytes4[](1);
+        _selectors[0] = TokenRateLimiter.updateTotalLimit.selector;
```

Comment:
Will we have 14D delay on `grantRole` (if we need to add rate limiter to new contracts)?

Reply:
Yes, if we don't need to delay for all the functions. Maybe we should make the owner as Scroll Multisig instead of the `ScrollOwner` contract.